libraries/command_line_tools.rb

```diff
@@ -0,0 +1,20 @@
+include Chef::Mixin::ShellOut
+
+module MacOS
+  class CommandLineTools
+    attr_reader :product
+
+    def initialize
+      @product = available_products.lines
+                                   .select { |s| s.include?('* Command') }
+                                   .last.tr('*', '').strip
+    end
+
+    def available_products
+      shell_out('softwareupdate -l').stdout
```

Comment:
Nit: Break the command into separate arguments and use the long version of the `list` option?

```ruby
shell_out('softwareupdate', '--list').stdout
```

Reply:
Yep.